### DISCUSSION TITLE
Is there any Automatic Annotation sample or guideline using my own model?

### DISCUSSION THREAD

**Discussion by rogerhsu1121** [cvat-ai/cvat#3074] [General]
I install cvat and nuclio on my local machine. 
I can run automatic annotation by the model provided by nuclio. 
But I wanna use my own model to do automatic annotation. 
How could I do that? 
I can see there is a guideline using onepanel, 
https://docs.onepanel.ai/docs/getting-started/quickstart/

But I wanna use my own model on cvat.
Is there any Automatic Annotation sample or guideline using custom model?

**Comment by azhavoro** [comment]
Hi, you need to implement a new serverless function and at the moment there is no detailed guide on how to do this, but I would suggest looking into the current implementations of one of the serverless functions as a reference https://github.com/openvinotoolkit/cvat/tree/develop/serverless.

**Reply by datscione** [reply]
This CVAT serverless tutorial may be helpful. https://openvinotoolkit.github.io/cvat/docs/manual/advanced/serverless-tutorial/

**Comment by bdlab-tech** [comment]
update the query:
Is there any guidance on how to retrain a model for automatic annotation provided by nuclio after running multiple annotations (whether manual or automatic) in CVAT? Or even use your own model instead of the base one from nuclio for automatic annotation.

**Comment by NamalJayasuriya** [comment]
Hi All,
I followed CVAT serverless tutorial and was able to try automatic anotations with models under openvino. I have my own model build with detectron2. When I tried their examples under pytorch non of them worked for me. Please let me know it it worked for anyone.
